cmd/pdcli/alerts/cmd.go

```diff
@@ -0,0 +1,298 @@
+		incidentAlerts, err = getIncidentAlerts(connection, id)
+
+		if err != nil {
+			return err
+		}
+
+		alerts = append(alerts, incidentAlerts...)
+	}
+
+	printAlerts(alerts)
+
+	return nil
+}
+
+// getIncidents returns a string slice consisting IDs of the first 10 incidents.
+func getIncidents(client *pagerduty.Client, isHigh bool) ([]string, error) {
+
+	var incidentIDs []string
+	var status []string
+	var teams []string
+	var users []string
+
+	var opts pagerduty.ListIncidentsOptions
+
+	var aerr pagerduty.APIError
+
+	// Get current user details
+	user, err := client.GetCurrentUser(pagerduty.GetCurrentUserOptions{})
+
+	if err != nil {
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+	}
+
+	// Check the assigned-to flag
+	switch args.assignment {
+
+	case "team":
+		// Fetch incidents belonging to a specific team
+		opts.TeamIDs = append(teams, constants.TeamID)
+
+	case "silentTest":
+		// Fetch incidents assigned to silent test
+		opts.UserIDs = append(teams, constants.SilentTest)
+
+	case "self":
+		// Fetch incidents only assigned to self
+		opts.UserIDs = append(users, user.ID)
+	}
+
+	// Fetch only triggered, acknowledged incidents (not resolved ones)
+	opts.Statuses = append(status, constants.StatusTriggered, constants.StatusAcknowledged)
+
+	// Let the number of incidents fetched
+	opts.Limit = constants.AlertsLimit
+
+	// Check if flag is true
+	if args.low {
+		opts.Urgencies = []string{"low"}
+	} else if args.high {
+		opts.Urgencies = []string{"high"}
+	}
+
+	eps, err := client.ListIncidentsWithContext(context.Background(), opts)
+
+	if err != nil {
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+	}
+
+	// Check if there are no incidents returned
+	if len(eps.Incidents) == 0 {
+		fmt.Println("Currently there are no alerts assigned to " + args.assignment)
+		os.Exit(1)
+	}
+
+	for _, incident := range eps.Incidents {
+		incidentIDs = append(incidentIDs, incident.Id)
+	}
+
+	return incidentIDs, nil
+}
+
+// getIncidentAlerts returns all the alerts belong to a particular incident.
+func getIncidentAlerts(client *pagerduty.Client, incidentID string) ([]alert, error) {
+
+	var alerts []alert
+
+	var opts pagerduty.ListIncidentAlertsOptions
+
+	eps, err := client.ListIncidentAlertsWithContext(context.Background(), incidentID, opts)
+
+	if err != nil {
+		var aerr pagerduty.APIError
+
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+
+	}
+
+	for _, p := range eps.Alerts {
+
+		tempAlertObj := alert{}
+
+		tempAlertObj.incidentID = incidentID
+		tempAlertObj.name = p.Summary
```

Comment:
should we be using the data that is closer to the source i.e. alert.body? This way we would be directly interacting with alert as it was sent by the cluster

Reply:
Each alert is tied to a particular incident, that's how I figured the PagerDuty API works. Hence we'll have to interact with the incident first to fetch Status, ID, Severity etc.

From what I see the alert.body doesn't contain those details.
```
"body": {
                "contexts": [],
                "details": {
                    "cluster_id": "48711345-283c-4cef-a901-5c63a326725a",
                    "component": "console-ErrorBudgetBurn",
                    "console": "https://console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org",
                    "firing": "Labels:\n - alertname = console-ErrorBudgetBurn\n - long_window = 1h\n - namespace = openshift-route-monitor-operator\n - probe_url = console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org\n - prometheus = openshift-monitoring/k8s\n - severity = critical\n - short_window = 5m\nAnnotations:\n - message = High error budget burn for console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org (current value: 1)\nSource: https://prometheus-k8s-openshift-monitoring.apps.nshneor-aug10.4v4e.i1.devshift.org/graph?g0.expr=1+-+%28sum%28sum_over_time%28probe_success%7Bprobe_url%3D%22console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org%22%7D%5B5m%5D%29%29+%2F+sum%28count_over_time%28probe_success%7Bprobe_url%3D%22console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org%22%7D%5B5m%5D%29%29%29+%3E+%2814.4+%2A+%281+-+0.995%29%29+and+sum%28count_over_time%28probe_success%7Bprobe_url%3D%22console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org%22%7D%5B5m%5D%29%29+%3E+5+and+1+-+%28sum%28sum_over_time%28probe_success%7Bprobe_url%3D%22console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org%22%7D%5B1h%5D%29%29+%2F+sum%28count_over_time%28probe_success%7Bprobe_url%3D%22console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org%22%7D%5B1h%5D%29%29%29+%3E+%2814.4+%2A+%281+-+0.995%29%29+and+sum%28count_over_time%28probe_success%7Bprobe_url%3D%22console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org%22%7D%5B1h%5D%29%29+%3E+60&g0.tab=1\n",
                    "group": "console-ErrorBudgetBurn",
                    "link": "https://github.com/openshift/ops-sop/tree/master/v4/alerts/console-ErrorBudgetBurn.md",
                    "link2": "",
                    "num_firing": "1",
                    "num_resolved": "0",
                    "resolved": ""
                },
```

Reply:
it's not immediately obvious, but there's lots of good information in `.firing`:
```
 - alertname = console-ErrorBudgetBurn
 - long_window = 1h
 - namespace = openshift-route-monitor-operator
 - probe_url = console-openshift-console.apps.nshneor-aug10.4v4e.i1.devshift.org
 - prometheus = openshift-monitoring/k8s
 - severity = critical
 - short_window = 5m
Annotations:
 - message = High error budget burn for console-openshift-console.apps..devshift.org (current value: 1)
Source: https://prometheus-k8s-openshift-monitoring.apps.nshneor-aug10.4v4e.i1.devshift.org/graph?g0.expr=...
" 
```
 I just wish it was json :(  perhaps something that could be fixed in alert manager.  Let's park this idea for now, but I'd really like to revisit.